torch/csrc/autograd/python_function.cpp

```diff
@@ -736,6 +734,15 @@ PyObject* process_outputs(PyObject *op_obj, THPFunction* grad_fn, const Unpacked
 
   grad_fn->cdata.num_inputs = num_outputs;
 
+  // Record type, device, and size information about inputs
+  if (grad_fn->cdata.is_executable) {
+    grad_fn->input_info = new std::vector<VariableInfo>();
+    grad_fn->input_info->reserve(unpacked.input_vars.size());
+    for (auto& var : unpacked.input_vars) {
+      grad_fn->input_info->emplace_back(var);
+    }
```

Comment:
This seems to have input and output inverted compared to the line above. There, `num_inputs = num_outputs`, so the attributes are named from the perspective of a backward function. Here, you use save inputs saying that they're inputs?

Reply:
Yeah, just like the for-loop below.